aes-gcm/src/lib.rs

```diff
@@ -170,27 +198,33 @@ pub struct AesGcm<Aes, NonceSize> {
 
     /// Length of the nonce.
     nonce_size: PhantomData<NonceSize>,
+
+    /// Length of the tag.
+    tag_size: PhantomData<TagSize>,
 }
 
-impl<Aes, NonceSize> KeySizeUser for AesGcm<Aes, NonceSize>
+impl<Aes, NonceSize, TagSize> KeySizeUser for AesGcm<Aes, NonceSize, TagSize>
 where
     Aes: KeySizeUser,
+    TagSize: self::TagSize,
```

Comment:
Kind of surprised these pass the `unused_qualifier` lint. Seems like they should just be:

```suggestion
    TagSize: TagSize,
```

Reply:
Yeah I added the `self::` because the generic type `TagSize` and the trait `TagSize` have name collisions.
So I had to add the `self::`, unless you want to change their names of course.

Reply:
Removing `self::` should be fine?

Unless you’re saying it’s confusing?

We can hopefully change them into const generic parameters after migrating to `hybrid-array` in the next release

Reply:
The compiler throws a compile error after removing `self::`.

```
error[E0404]: expected trait, found type parameter `TagSize`
   --> aes-gcm\src\lib.rs:191:14
    |
137 | pub trait TagSize: private::SealedTagSize {}
    | -------------------------------------------- you might have meant to refer to this trait
...
189 | pub struct AesGcm<Aes, NonceSize, TagSize = U16>
    |                                   ------- found this type parameter
190 | where
191 |     TagSize: TagSize,
    |              ^^^^^^^ not a trait
```

Reply:
Aah, ok